### ISSUE TITLE
RSA key verify error

### ISSUE THREAD

**Issue by rhorer** [sidneys/homebrew-homebrew#23]
Installing with Homebrew:

==> Installing chntpw from sidneys/homebrew
==> Installing dependencies for sidneys/homebrew/chntpw: sidneys/homebrew/openssl@1.0
==> Installing sidneys/homebrew/chntpw dependency: sidneys/homebrew/openssl
==> Patching
==> Applying openssl-1.0.2u-darwin-arm64.patch
patching file Configure
==> perl ./Configure --prefix=/opt/homebrew/Cellar/openssl@1.0/1.0.2u_1 --openss
==> make
==> make test
Last 15 lines from /Users/krhorer/Library/Logs/Homebrew/openssl@1.0/03.make:
g = 2
Salt = 645A128752D3DA9FA1D9DA7F6386089A3D4B3F5A
Verifier = ED4054705449F1AF24C5EFD522E60B839523B70CAFA97AF60E41532CAA84B78F3EF3EF2521F892C78FA44439AD10163E4B5551CEB4767CD047948129CC4F9619E6DAEB34EDF34330AFB01EA15B68A2AD20265B57D1E32D53BA628D0062A0E9D2AF72D21BCFA89186961A5E7178857E3CFB81BC6CBA74FD4624C764C0B2DD9186
b = A778C1F93515B41A6DD135DD3A373D40BBDD2269B1293FA7A671821157016BA2
B = 7F12B4670749158081F222C464E75722BA9448055223AA464777FA6EDEAD2C6259E7C6E176C6E9A85F79E7E776327C4B100B75502AF6E7882A2F3B5351A2943CD47D2C6E44434D0A18F7CF081638C00E9989C7F4C8512AC51BA95F4359B3B164BB6FDACE3A5E3F0F0A2D8CFE0D7C9B0BB9C1CE820A763DB6F4E140804360B06C
a = B31C6CCDAF6EF77F970482580214E6FC3D7090ED61D6D53A295702943FB76057
A = 9CCD63FE20077E1F754D724AF099A7DD950E0433259F77F838C6B67E77079272CD7DBA69E1A5692DEEDCB75298FBAD72CB760AE1E9519CA562CDC3176538B69D75F909127CDBD3C6FFA060E1B75266F370BC91DAA8F3ED4DFDB984E96E09A6F97DBA3B823727F5CDE29E1C54C3B4F496CB7D41A341F29FF3437DAB8995AD312
Client's key = A7BF77723E94258147DCCACBB4E4B7AB10200A10E784AD2E6864F05638F79831E903193EB5F25F11843F71AB1B1F936F509798EB4DAFD5EA022CB5B95A1D0A6E1A4EAB42C6526474A483A7860848446DBB9B6B0F2D0BEED7C1977682AE7F378FBFA5AACA39FA63E96AC9C119F189E3EEA60F27AAE785427C391EEA23BF76BCD4
Server's key = A7BF77723E94258147DCCACBB4E4B7AB10200A10E784AD2E6864F05638F79831E903193EB5F25F11843F71AB1B1F936F509798EB4DAFD5EA022CB5B95A1D0A6E1A4EAB42C6526474A483A7860848446DBB9B6B0F2D0BEED7C1977682AE7F378FBFA5AACA39FA63E96AC9C119F189E3EEA60F27AAE785427C391EEA23BF76BCD4
CMS consistency test
/usr/bin/perl cms-test.pl
CMS => PKCS#7 compatibility tests
signed content DER format, RSA key: verify error
make[1]: *** [test_cms] Error 1
make: *** [tests] Error 2

**Comment by rhorer** [comment]
I should have mentioned that I'm on an Apple Silicon M1 MacBook Pro running 12.6.7
[03.make.log](https://github.com/sidneys/homebrew-homebrew/files/12045076/03.make.log)

**Comment by florianlouvet** [comment]
Anyone has as solution for this? Seems to get the same error here.

**Comment by Qwertylex** [comment]
I have the same error as well.

**Comment by DevelCL** [comment]
Idem, same error for 1.0.2k release.

**Comment by ladar** [comment]
Yes, I have a fix. The issue is minor. The OpenSSL unit tests fail because some of the certificates embedded in the tarball have expired. 

As a short term workaround you coi;fd always reset your system clock to Jan 1st 2023 and try the install again. I haven't tested this myself, but in theory, if your clock isn't syncing with the internet, it should work.

Long term I adapted the 1.1.1 patch for 1.0..2 and you can find it here:

[1.0.2_update_expiring_certificates.patch](https://raw.githubusercontent.com/lavabit/magma/682101e08114be9b006aadb228943c487dfb1abf/lib/patches/openssl/1.0.2_update_expiring_certificates.patch)

@rhorer @florianlouvet @Qwertylex @DevelCL can one of you take charge of getting the fix into Homebrew?

**Comment by hellysmile** [comment]
Thnx @ladar 

I had same issue and prepared a patched version brew Formula

https://gist.github.com/hellysmile/f58a6e32656ac1101cf6a64d0674f2c8 with openssl 1.0.2 patch from https://github.com/lavabit/magma
